Sources/Fluent/Schema/Schema+Creator.swift

```diff
@@ -134,12 +135,34 @@ extension Schema {
         ) {
             fields += Field(
                 name: "\(entity.name)_id",
-                type: .int,
+                type: E.idType?.fieldType ?? .int,
                 optional: optional,
                 unique: unique,
                 default: `default`
             )
         }
+        
+        public func pivotKeys(
+            left: Entity.Type,
+            right: Entity.Type,
+            suffix: String
+        ) {
+            let leftKeyType = left.idType ?? Schema.Field.KeyType.int
+            let rightKeyType = right.idType ?? Schema.Field.KeyType.int
+            
+            let leftName = "\(left.name)\(suffix)"
+            let rightName = "\(right.name)\(suffix)"
+            
+            fields += Field(name: leftName,
+                            type: leftKeyType.fieldType,
+                            optional: false,
+                            unique: false)
+            
+            fields += Field(name: rightName,
+                            type: rightKeyType.fieldType,
```

Comment:
@mz2 I'm asking if there's potentially a mistake here with the fieldType since normally, it corresponds to an `idType`, but I might be misunderstanding the role within a pivot, so just wanted to clarify :)